packages/core-v2/src/entity-model/async-queryable/implementation.ts

```diff
@@ -0,0 +1,78 @@
+import {AsyncQueryableEntityModel, AsyncQueryableObservableEntityModel} from "./model";
+import {Entity} from "../entity";
+import {InMemoryEntityModel} from "../implementation";
+
+export class SimpleAsyncQueryableObservableEntityModel extends InMemoryEntityModel implements AsyncQueryableObservableEntityModel {
+    protected model: AsyncQueryableEntityModel;
+    protected queries: Record<string, any> = {};
+    protected reverseQueries: Map<string, string[]> = new Map();
+
+    protected addToReverseQueries(entityIri: string, queryIri: string): void {
+        if (!this.reverseQueries.has(entityIri)) {
+            this.reverseQueries.set(entityIri, []);
+        }
+        const queries = this.reverseQueries.get(entityIri)!;
+        if (!queries.includes(queryIri)) {
+            queries.push(queryIri);
+        }
+    }
+
+    constructor(queryableModel: AsyncQueryableEntityModel) {
+        super();
+        this.model = queryableModel;
+    }
+
+    async addQuery(queryIri: string): Promise<void> {
+        if (this.queries[queryIri]) {
+            // Do nothing, because the query is present
+            return;
+        }
+        const result = await this.model.query(queryIri);
+        if (this.queries[queryIri]) {
```

Comment:
Proc je tohle tady dvakrat, zde a na radku 26?

Reply:
Na řádku 30 dělám await, takže v mezičase někdo jiný to tam může ten query přidat. Proto to zkontroluji znova. Ten kód se dá napsat optimálněji, zatím to je jen proof-of-concept.